### PR TITLE
fix: Fix issues parsing and displaying commitments

### PR DESCRIPTION
**Pull request by rajivshah3**
- Minor refactoring
- Fixes credential parsing for commitments (they were not being mapped properly into the store)
- Adjust UI to represent commitments
- Adjust commitment logo styling

### CODE REVIEW

File: ui/views/CredentialInfo.svelte [hunk 1]
```diff
@@ -229,10 +231,19 @@
...
-                        <li>
-                            <p>{object.key}</p>
-                            <span>{object.value}</span>
-                        </li>
+                        {#if object.key[0] && object.key[0] === 'commitments'}
+                            {#each object.value as commitment}
+                                <li>
+                                    <p>{commitment.commitmentTitle} ({commitment.commitmentSupport})</p>
+                                    <span>{commitment.commitmentPercentage}</span>
+                                </li>
```
Comment (rajivshah3) [inline ui/views/CredentialInfo.svelte:239]: @sobolev-alexey I'm happy to change this part if you want to adjust how commitment info is displayed

Comment (sobolev-alexey) [reply ui/views/CredentialInfo.svelte:239]: This part is a little bit tricky. 
Please add a sub-condition to check which commitment type it is, then you can display the commitment using the following logic:

commitment === `FutureCommitment`
```
 <p>
       {commitment?.condition} {commitment?.percentage}% 
       THEN donate {commitment?.walletPercentage}% of my wallet balance TO support {commitment?.support}
</p>
```
commitment === `PresentCommitment`
```
  <p>I commit to {commitment?.support}</p>
```

File: ui/lib/identity/index.ts [hunk 2]
```diff
@@ -150,25 +150,25 @@ export type InsuranceData = {
...
-    commitmentId: string;
-    commitmentTitle: string;
-    commitmentPercentage: number;
-    commitmentSupport: string;
-    commitmentWalletPercentage: number;
+    CommitmentId: string;
+    CommitmentTitle: string;
+    CommitmentPercentage: number;
+    CommitmentSupport: string;
+    CommitmentWalletPercentage: number;
```
Comment (sobolev-alexey) [inline ui/lib/identity/index.ts:157]: could you please add another parameter to the both Commitment types:
```
CommitmentCondition: string;
```

File: ui/lib/helpers.ts [hunk 3]
```diff
@@ -325,7 +325,13 @@ export const prepareInsuranceInformation = (insuranceData: InsuranceData): Insur
  * @returns {FutureCommitmentInfo}
  */
 export const prepareFutureCommitmentInformation = (futureCommitmentData: FutureCommitmentData): FutureCommitmentInfo => ({
-    commitments: futureCommitmentData.commitments
+    commitments: futureCommitmentData.Commitments.map((commitment) => ({
+        commitmentId: commitment.CommitmentId,
+        commitmentTitle: commitment.CommitmentTitle,
+        commitmentPercentage: commitment.CommitmentPercentage,
+        commitmentSupport: commitment.CommitmentSupport,
+        commitmentWalletPercentage: commitment.CommitmentWalletPercentage
```
Comment (sobolev-alexey) [inline ui/lib/helpers.ts:333]: could you please add another parameter to the both Commitment types:
```
commitmentCondition: commitment.CommitmentCondition,
```

File: ui/lib/helpers.ts [hunk 4]
```diff
@@ -338,7 +344,13 @@ export const prepareFutureCommitmentInformation = (futureCommitmentData: FutureC
  * @returns {PresentCommitmentInfo}
  */
 export const preparePresentCommitmentInformation = (presentCommitmentData: PresentCommitmentData): PresentCommitmentInfo => ({
-    commitments: presentCommitmentData.commitments
+    commitments: presentCommitmentData.Commitments.map((commitment) => ({
+        commitmentId: commitment.CommitmentId,
+        commitmentTitle: commitment.CommitmentTitle,
+        commitmentPercentage: commitment.CommitmentPercentage,
+        commitmentSupport: commitment.CommitmentSupport,
+        commitmentWalletPercentage: commitment.CommitmentWalletPercentage
```
Comment (sobolev-alexey) [inline ui/lib/helpers.ts:352]: could you please add another parameter to the both Commitment types:
```
commitmentCondition: commitment.CommitmentCondition,
```

**Review by sobolev-alexey — CHANGES_REQUESTED** [review]

**Review by sobolev-alexey — APPROVED** [review]

**Pull request merged**